src/coreclr/jit/utils.cpp

```diff
@@ -1520,13 +1520,16 @@ void HelperCallProperties::init()
         bool mutatesHeap   = false; // true if any previous heap objects [are|can be] modified
         bool mayRunCctor   = false; // true if the helper call may cause a static constructor to be run.
         bool isNoEscape    = false; // true if none of the GC ref arguments can escape
+        bool isNoGC        = false; // true is the helper cannot trigger GC
 
         switch (helper)
         {
             // Arithmetic helpers that cannot throw
             case CORINFO_HELP_LLSH:
             case CORINFO_HELP_LRSH:
             case CORINFO_HELP_LRSZ:
+                isNoGC = true;
+                FALLTHROUGH;
             case CORINFO_HELP_LMUL:
             case CORINFO_HELP_LNG2DBL:
             case CORINFO_HELP_ULNG2DBL:
```

Comment:
just randomly checked `CORINFO_HELP_ULNG2DBL` and it also looks nogc, but I presume your PR just cleans things up

Reply:
Right, this is intentionally meant to be no-diff. Additionally, these no-gc annotations are a bit scary to maintain: they must match between CoreCLR and NAOT, and only one of those is tested with GCStress.